pulp_ansible/app/serializers.py

```diff
@@ -361,151 +373,298 @@ class Meta:
 
 class CollectionVersionUploadSerializer(SingleArtifactContentUploadSerializer):
     """
-    A serializer for CollectionVersion Content.
+    A serializer with the logic necessary to upload a CollectionVersion.
+
+    Used in ``.viewsets.CollectionVersionViewSet`` and ``.galaxy.v3.views.CollectionUploadViewSet``
+    to perform the creation and validation of a CollectionVersion and add to repository if
+    necessary. This serializer is meant to be compliant with
+    ``pulpcore.plugin.viewsets.SingleArtifactContentUploadViewSet`` and thus follows these steps on
+    creation:
+
+    1. ``SingleArtifactContentUploadViewSet.create()`` calls ``validate()`` with request in context
+    2. Task payload is created, converting uploaded-file (if present) to an artifact
+    3. ``pulpcore.plugin.tasks.general_create`` is dispatched with this serializer, the task payload
+        and a deferred context determine by the viewset (default an empty context).
+    4. ``general_create`` calls ``validate()`` again with deferred context now.
+        ``deferred_validate()`` is now called and the upload object (if present) is converted to an
```

Comment:
Isn't this where we also call the import_collection to extract all the metadata? That process that may be offloaded into a different application cluster?

Reply:
Yes, I wrote these steps describing how the pulpcore machinery works since I feel it is not clear from just looking at the serializer what is going on.